src/sentry/api/serializers/models/group.py

```diff
@@ -183,19 +182,24 @@ def __init__(
         self.collapse = collapse
         self.expand = expand
 
-    def _serialize_assigness(
-        self, actor_dict: Mapping[int, ActorTuple]
-    ) -> Mapping[int, Union[Team, Any]]:
-        actors_by_type: MutableMapping[Any, List[ActorTuple]] = defaultdict(list)
-        for actor in actor_dict.values():
-            actors_by_type[actor.type].append(actor)
+    def _serialize_assignees(self, item_list: Sequence[Group]) -> Mapping[int, Union[Team, Any]]:
+        gas = GroupAssignee.objects.filter(group__in=item_list)
+        result: MutableMapping[int, Union[Team, Any]] = {}
+        all_team_ids: MutableMapping[int, int] = {}
+        all_user_ids: MutableMapping[int, int] = {}
 
-        resolved_actors = {}
-        for t, actors in actors_by_type.items():
-            serializable = ActorTuple.resolve_many(actors)
-            resolved_actors[t] = {actor.id: actor for actor in serializable}
+        for g in gas:
+            if g.team_id:
+                all_team_ids[g.team_id] = g.group_id
+            if g.user_id:
+                all_user_ids[g.user_id] = g.group_id
```

Comment:
Do teams only have a single group? If `item_list` is a collection of groups, couldn't a team be assigned to multiple groups?

Reply:
Yes, I think you may be right.  I'll take a closer look to see if this can be simplified.  In general, I wish we had a much more robust hybrid cloud actor interface.